File: Oidc.AuthorizationServer/Models/UserClaim.cs
using FluentNHibernate.Data;
using lvl.Ontology;
using lvl.Ontology.Authorization;
using lvl.Ontology.Conventions;
using System.ComponentModel.DataAnnotations;

namespace lvl.Oidc.AuthorizationServer.Models
{
    /// <summary>
    ///     Description about who a user is.
    /// </summary>
    [Schema("oidc")]
    [HiddenFromApi]
    public class UserClaim : Entity, IAggregateRoot
    {
        /// <summary>
        ///     Description about the identity of the user.
        /// </summary>
        [Required]
        public string Name { get; set; }
    }
}

File: oidc/src/models/user-claim.ts
import { Entity, IAggregateRoot, Required } from '@lvl/front-end';

export class UserClaim extends Entity implements IAggregateRoot {
    @Required() public name: string;

    constructor(options?: IUserClaimOptions) {
        options = options || {};
        super(options);
        this.name = options!.name;
    }
}

interface IUserClaimOptions {
    name?: string;
    id?: number;
}
